sycl/include/CL/sycl/device.hpp

```diff
@@ -100,8 +101,8 @@ class device {
   std::shared_ptr<detail::device_impl> impl;
   device(std::shared_ptr<detail::device_impl> impl) : impl(impl) {}
 
-  template <class T>
-  friend decltype(T::impl) detail::getSyclObjImpl(const T &SyclObject);
+  template <class Obj>
```

Comment:
It this change related to "Fix "getSyclObjImpl is not a function" error."?
Could you add a comment preventing "Obj" -> "T", please?
If I recall correctly it's a workaround for some MSVC specific bug. Right?

Reply:
Yes, that is a workaround for MSVC bug. I added a comment in detail/common.hpp